ml/README.md

```diff
@@ -0,0 +1,43 @@
+# spark-cuml
+
+RAPIDS ML - RAPIDS Machine Learning Library on Spark
+
+
+## API change
+
+We describe main API changes for gpu accelerated algorithms:
+
+### 1. PCA
+```scala
+val pca = new org.apache.spark.ml.feature.PCA()
+  .setInputCol("feature")
+  .setOutputCol("feature_value_3d")
+  .setK(3)
+  .useGemm(true) // or false, switch to use original BLAS bsr or cuBLAS gemm to compute covariance matrix
+  .meanCentering(true) // or false, switch to do mean centering before computing covariance matrix
+  .fit(vectorDf)
+```
+
+## Build
+The artifact jar for this submodule will not be included in the plugin jar of the whole spark-rapids project. User can build it directly in the _project root path_:
+```
+mvn -pl ml -am clean package
+```
+Then `rapids-4-spark-ml_2.12-21.10.0-SNAPSHOT.jar` will be generated under `ml/target` folder.
+
+_Note_: This module contains both native and Java/Scala code. When compiling native code to get the essential library, `cmake`, `ninja`, and `gcc` are all required. 
```

Comment:
perhaps more instructions on how to build native part. do I run a separate command, does it run as part of the mvn?  If part of mvn what happens if I don't have these installed?